daiquiri/__init__.py

```diff
@@ -51,21 +51,66 @@ def process(self, msg, kwargs):
         return msg, kwargs
 
 
+PARAMETER_TEMPLATE = ' [{0}: {1}]'
+
+
+class ArbitraryContextAdapter(logging.LoggerAdapter):
+    """Logger adapter to add context string to log record's extra data
+
+    Keywords passed to the log call are formatted into a context
+    string, which is then added to the "extra" dictionary passed to
+    the underlying logger so they are emitted with the log message and
+    available to the format string.
+
+    Example:
+      A formatter with the format string of "%(context)s %(messages)"
+      called like logger.info("A message.", test1="a", test2="b")
+      would log the message: " [test1: a] [test2: b] A message."
+
+    Special keywords:
+
+    extra
+      An existing dictionary of extra values to be passed to the
+      logger. If present, the dictionary is copied and extended.
+    """
+
+    def process(self, msg, kwargs):
+        # Make a new extra dictionary combining the values we were
+        # given when we were constructed and a 'context' key which
+        # will contain a formatted string containing anything from
+        # kwargs.
+        extra = self.extra.copy()
+        if 'extra' in kwargs:
+            extra.update(kwargs.pop('extra'))
+        # Format any unknown keyword arguments into the context string.
+        context = ''
+        for name in list(kwargs.keys()):
+            if name == 'exc_info':
+                continue
+            context += PARAMETER_TEMPLATE.format(name, kwargs.pop(name))
+        extra['context'] = context
+        extra['_daiquiri_extra'] = extra
+        kwargs['extra'] = extra
+        return msg, kwargs
```

Comment:
Haha, right, don't go that way indeed. I love how you destroyed my idea with good arguments, thanks. 😁

What'd be a better way to do it if we allow ourselves to break the API?

I'd really love to have a `%(rest-of-extra)` but clearly that does not seem easily/safely possible. So if we're left with `%(context)`, what I'm scared of is that it'll be less obvious to use it since users need to use a different adapter. I'm sure that'll confuse a lot of people.

(Just reasserting that – I'm not trying to discuss this patch to annoy you as you can imagine, I'm just trying to find a more user friendly way than passing a class to `getLogger` 🤓)